File: src/main/java/com/google/android/finsky/api/model/DfeList.java
package com.google.android.finsky.api.model;

import com.android.volley.Request;
import com.anod.appwatcher.utils.CollectionsUtils;
import com.google.android.finsky.api.DfeApi;
import com.google.android.finsky.api.DfeUtils;
import com.google.android.finsky.protos.nano.Messages;
import com.google.android.finsky.protos.nano.Messages.ListResponse;

import java.util.Arrays;
import java.util.List;


public final class DfeList extends ContainerList<ListResponse>
{
    private final DfeApi mDfeApi;

    public DfeList(final DfeApi dfeApi, final String mInitialListUrl, final boolean autoLoadNextPage, CollectionsUtils.Predicate<Document> responseFilter) {
        super(mInitialListUrl, autoLoadNextPage, responseFilter);
        mDfeApi = dfeApi;
    }

    @Override
    protected final Request<?> makeRequest(final String url) {
        return this.mDfeApi.list(url, this, this);
    }

}

File: src/main/java/com/google/android/finsky/api/model/DfeList.kt
package com.google.android.finsky.api.model

import com.android.volley.Request
import com.google.android.finsky.api.DfeApi
import com.google.android.finsky.protos.nano.Messages.ListResponse


class DfeList(private val dfeApi: DfeApi,
              initialListUrl: String,
              autoLoadNextPage: Boolean,
              filter: ((Document?) -> Boolean))
    : ContainerList<ListResponse>(initialListUrl, autoLoadNextPage, filter) {

    override fun makeRequest(url: String): Request<*> {
        return this.dfeApi.list(url, this, this)
    }

}